costs/src/storage_cost/mod.rs

```diff
@@ -0,0 +1,86 @@
+use std::ops::{Add, AddAssign};
+
+use crate::{
+    error::Error,
+    storage_cost::removal::{StorageRemovedBytes, StorageRemovedBytes::NoStorageRemoval},
+};
+
+/// Key Value Storage Costs
+pub mod key_value_cost;
+/// Storage Removal
+pub mod removal;
+/// Costs to Transitions
+pub mod transition;
+
+/// Storage only Operation Costs
+#[derive(Debug, PartialEq, Clone, Eq)]
+pub struct StorageCost {
+    /// How many bytes are said to be added on hard drive.
+    pub added_bytes: u32,
+    /// How many bytes are said to be replaced on hard drive.
+    pub replaced_bytes: u32,
+    /// How many bytes are said to be removed on hard drive.
+    pub removed_bytes: StorageRemovedBytes,
+}
+
+impl Add for StorageCost {
+    type Output = Self;
+
+    fn add(self, rhs: Self) -> Self::Output {
+        Self {
+            added_bytes: self.added_bytes + rhs.added_bytes,
+            replaced_bytes: self.replaced_bytes + rhs.replaced_bytes,
+            removed_bytes: self.removed_bytes + rhs.removed_bytes,
+        }
+    }
+}
+
+impl AddAssign for StorageCost {
+    fn add_assign(&mut self, rhs: Self) {
+        self.added_bytes += rhs.added_bytes;
+        self.replaced_bytes += rhs.replaced_bytes;
+        self.removed_bytes += rhs.removed_bytes;
+    }
+}
+
+impl StorageCost {
+    /// Verify that the len of the item matches the given storage_cost cost
+    pub fn verify(&self, len: u32) -> Result<(), Error> {
+        let size = self.added_bytes + self.replaced_bytes;
+        match size == len {
```

Comment:
that's some extremely fancy type of `if`

Reply:
Indeed, fixed